src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -392,52 +457,93 @@ private static CodeActionKind GetCodeActionKindFromSuggestedActionCategoryName(s
                 _ => throw ExceptionUtilities.UnexpectedValue(priority)
             };
 
-        public static CodeAction? GetCodeActionToResolve(string distinctTitle, ImmutableArray<CodeAction> codeActions)
+        public static CodeAction? GetCodeActionToResolve(string[] codeActionPath, ImmutableArray<CodeAction> codeActions, bool isFixAllAction)
         {
-            // Searching for the matching code action. We compare against the unique identifier
-            // (e.g. "Suppress or Configure issues|Configure IDExxxx|Warning") instead of the
-            // code action's title (e.g. "Warning") since there's a chance that multiple code
-            // actions may have the same title (e.g. there could be multiple code actions with
-            // the title "Warning" that appear in the code action menu if there are multiple
-            // diagnostics on the same line).
-            foreach (var codeAction in codeActions)
+            var currentActions = codeActions;
+            for (var i = 0; i < codeActionPath.Length; i++)
             {
-                var action = CheckForMatchingAction(codeAction, distinctTitle);
-                if (action != null)
-                    return action;
+                var title = codeActionPath[i];
+                var matchingActions = currentActions.Where(action => action.Title == title);
+                CodeAction? matchingAction = null;
+
+                // If we only have one matching action then just need to retrieve it from the list.
+                if (matchingActions.Count() == 1)
+                {
+                    matchingAction = matchingActions.FirstOrDefault();
+                }
+                else
+                {
+                    // Otherwise, we are likely at the end of the path and need to retrieve
```

Comment:
could update this comment to mention that we do this because both the fixall and regular code action have the same title